android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -238,6 +240,17 @@ public List<AccountInfo> stripNoSwapSupportedAccounts(List<AccountInfo> accountI
         return accountInfosFiltered;
     }
 
+    public void getKeyringPerId(String keyringId, Callbacks.Callback1<KeyringInfo> callback) {
+        if (TextUtils.isEmpty(keyringId)) return;
+        KeyringInfo selectedCoinKeyring = _mSelectedCoinKeyringInfoLiveData.getValue();
+        if (selectedCoinKeyring != null && selectedCoinKeyring.id.equals(keyringId)) {
+            callback.call(selectedCoinKeyring);
+        } else {
+            mKeyringService.getKeyringInfo(
```

Comment:
is it possible that `mKeyringService` is `null` in that place?

Reply:
I consider that scenario after looking at the crash though I believe the app should have crashed before this point if there was an issue with services in the domain models.
Alternately, we can wait and confirm if this gets fixed or I can add a null check if you like.

Reply:
let's better to do a null check if we are not confident. It would be ok to take a look on how it performs after the merge, but we want to uplift it, so I don't want we introduce that crash instead of what you fixed in that PR.

Reply:
done https://github.com/brave/brave-core/commit/4040133a8654cb1e23e618ee4031fb03869d1d5f